src/humancellatlas/data/metadata/api.py

```diff
@@ -773,15 +812,21 @@ def from_json_vx(core_cls: Type[E], **kwargs) -> MutableMapping[UUID4, E]:
         self.entities = {**self.projects, **self.biomaterials, **self.processes, **self.protocols, **self.files}
 
         links = metadata_files['links.json']['links']
-        self.links = list(chain.from_iterable(map(Link.from_json, links)))
+        schema_version = tuple(int(x) for x in metadata_files['links.json']['schema_version'].split('.'))
+        self.links = []
+        for link in links:
+            self.links.extend(Link.from_json(link, schema_version))
 
         for link in self.links:
             source_entity = self.entities[link.source_id]
             destination_entity = self.entities[link.destination_id]
-            assert isinstance(source_entity, LinkedEntity)
-            assert isinstance(destination_entity, LinkedEntity)
-            source_entity.connect_to(destination_entity, forward=True)
-            destination_entity.connect_to(source_entity, forward=False)
+            if hasattr(link, 'link_type') and link.link_type == 'supplementary_file_link':
+                pass
+            else:  # 'process_link'
+                assert isinstance(source_entity, LinkedEntity)
+                assert isinstance(destination_entity, LinkedEntity)
+                source_entity.connect_to(destination_entity, forward=True)
+                destination_entity.connect_to(source_entity, forward=False)
```

Comment:
I feel it would be more assertive if you put `# process_link` logic in the if conditional by negating the current one or other means. That way you could avoid the else branch. But it is up to you. 
I would also use a `continue` since there is nothing else in the loop that needs to be process after.

Reply:
Thanks. I've refactored the `if` condition to clean up the `for` loop.